ocaml/idl/api_errors.ml

```diff
@@ -188,6 +188,7 @@ let vm_memory_size_too_low = "VM_MEMORY_SIZE_TOO_LOW"
 let vm_memory_target_wait_timeout = "VM_MEMORY_TARGET_WAIT_TIMEOUT"
 let vm_shutdown_timeout = "VM_SHUTDOWN_TIMEOUT"
 let vm_duplicate_vbd_device = "VM_DUPLICATE_VBD_DEVICE"
+let vm_cant_set_dyn_memory = "VM_CANT_SET_DYN_MEMORY"
```

Comment:
If the goal is to restrict memory ballooning, then we already have an error for this: `memory_constraint_violation`.